.github/workflows/nightly.yml

```diff
@@ -0,0 +1,299 @@
+name: 🌒 Nightly Release
+
+on:
+  schedule:
+    - cron: "0 7 * * *" # every day at 12AM PST
+
+jobs:
+  nightly:
+    name: 🌒 Nightly Release
+    if: github.repository == 'remix-run/remix'
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🛑 Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: dev
+          token: ${{ secrets.NIGHTLY_PAT }}
+          fetch-depth: 0
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          cache: "yarn"
+
+      - name: 📥 Install deps
+        # even though this is called "npm-install" it does use yarn to install
+        # because we have a yarn.lock and caches efficiently.
+        uses: bahmutov/npm-install@v1
+
+      - name: 🏗 Build
+        run: yarn build
+
+      - name: ⤴️ Update Version
+        id: version
+        run: |
+          SHA=$(git rev-parse HEAD)
+          SHORT_SHA=${SHA::7}
+          NEXT_VERSION=0.0.0-nightly-${SHORT_SHA}
+          echo ::set-output name=NEXT_VERSION::${NEXT_VERSION}
+
+          git checkout -b nightly/${NEXT_VERSION}
+          npm version ${NEXT_VERSION}
+
+          git push origin --tags
+
+      - name: 🔐 Setup npm auth
+        run: |
+          echo "registry=https://registry.npmjs.org" >> ~/.npmrc
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> ~/.npmrc
+
+      - name: 🚀 Publish
+        run: npm run publish
+
+      - name: 🐱 Create GitHub release
+        uses: actions/create-release@v1
+        with:
+          draft: false
+          prerelease: true
+          release_name: v${{ steps.version.outputs.NEXT_VERSION }}
+          tag_name: v${{ steps.version.outputs.NEXT_VERSION }}
+        env:
+          # need this token in order to have it trigger the comment workflow
+          GITHUB_TOKEN: ${{ secrets.NIGHTLY_PAT }}
+
```

Comment:
Would love to figure out how to avoid the duplication between our release scripts 😬

Maybe something will turn up for this: https://twitter.com/kentcdodds/status/1513601107875885059

Reply:
It did!

Wanna look into putting all this post-release test stuff in its own workflow and use `npm hook` to kick it off?

https://docs.npmjs.com/cli/v8/commands/npm-hook

Reply:
yes! definitely - the main reason for the deployments in the first place was for pre-releases/experimental/nightlies 😉

Reply:
npm's webhook unfortantely wont work as they don't align with the body that github wants for their `repository_dispatch` events, but(!!) we can use a `needs: []` and manually curl the event to github and have that kick off the commenting 🎉  [https://github.com/mcansh/nightly-release-test/blob/main/.github/workflows/nightly.yml#L76-L98](https://github.com/mcansh/nightly-release-test/blob/main/.github/workflows/nightly.yml#L76-L98)

aside, but splitting the action into a nightly and a manual one, we need to check if the new `github.ref` in the manual action contains `nightly` and abort if so

Reply:
Oh, that's a great idea!